sdk/cosmos/README.md

```diff
@@ -1,27 +1,72 @@
+- [Prerequisites](#Prerequisites)
+- [Usage Code Sample](#Usage-Code-Sample)
+- [Guide for Production](#Guide-for-Production)
 - [FAQ](#faq)
-- [Release changes](#release-changes)
-- [Contribution and Feedback](#contribution-and-feedback)
+- [Release changes](#Release-changes)
+- [Contribution and Feedback](#Contribution-and-feedback)
 - [License](#license)
 
 <!-- /TOC -->
 
-## Example
+## Consuming the official Microsoft Azure Cosmos DB Java SDK
+
+This project provides a SDK library in Java for interacting with [SQL API](https://docs.microsoft.com/en-us/azure/cosmos-db/sql-api-sql-query) of [Azure Cosmos DB
+Database Service](https://azure.microsoft.com/en-us/services/cosmos-db/). This project also includes samples, tools, and utilities.
+
+Jar dependency binary information for maven and gradle can be found here at [maven](https://mvnrepository.com/artifact/com.microsoft.azure/azure-cosmos).
+
+For example, using maven, you can add the following dependency to your maven pom file:
+
+[//]: # ({x-version-update-start;com.microsoft.azure:azure-cosmos;current})
+```xml
+<dependency>
+  <groupId>com.microsoft.azure</groupId>
+  <artifactId>azure-cosmos</artifactId>
+  <version>3.4.0</version>
+</dependency>
+```
+[//]: # ({x-version-update-end})
+
+Useful links:
+
+- [Sample Get Started APP](https://github.com/Azure-Samples/azure-cosmos-db-sql-api-async-java-getting-started)
+- [Introduction to Resource Model of Azure Cosmos DB Service](https://docs.microsoft.com/en-us/azure/cosmos-db/sql-api-resources)
+- [Introduction to SQL API of Azure Cosmos DB Service](https://docs.microsoft.com/en-us/azure/cosmos-db/sql-api-sql-query)
+- [Reactor Core JavaDoc API](https://projectreactor.io/docs/core/release/api/)
+- [SDK FAQ](faq/)
+
+## Prerequisites
+
+- Java Development Kit 8
+- An active Azure account. If you don't have one, you can sign up for a [free account](https://azure.microsoft.com/free/). Alternatively, you can use the [Azure Cosmos DB Emulator](https://azure.microsoft.com/documentation/articles/documentdb-nosql-local-emulator) for development and testing. As emulator https certificate is self signed, you need to import its certificate to java trusted cert store as [explained here](https://docs.microsoft.com/en-us/azure/cosmos-db/local-emulator-export-ssl-certificates)
+- (Optional) SLF4J is a logging facade.
+- (Optional) [SLF4J binding](http://www.slf4j.org/manual.html) is used to associate a specific logging framework with SLF4J.
+- (Optional) Maven
+
+SLF4J is only needed if you plan to use logging, please also download an SLF4J binding which will link the SLF4J API with the logging implementation of your choice. See the [SLF4J user manual](http://www.slf4j.org/manual.html) for more information.
+
+<!-- TODO - update once JavaDoc is published
+## API Documentation
+
+Javadoc is available [here](https://azure.github.io/azure-cosmosdb-java/2.4.0/com/microsoft/azure/cosmosdb/rx/AsyncDocumentClient.html).
```

Comment:
This refers to javadoc of async v2.

Reply:
@kushagraThapar what's the reference link to use here?

Reply:
@kirankumarkolli It should be this: https://azuresdkdocs.blob.core.windows.net/$web/java/azure-cosmos/4.0.0-preview.1/index.html
Once we GA, we can update this link to GA version.